tests/plugins/image2/resizerwithreadonly.js

```diff
@@ -5,48 +5,61 @@
 	'use strict';
 
 	bender.test( {
-		// #719
+		// #719, #2816, #2874
 		'test readOnly set to false': function() {
 			bender.editorBot.create( {
-				name: 'editor1',
-				config: {
-					language: 'en'
-				}
+				name: 'editor1'
 			}, function( bot ) {
 				var editor = bot.editor;
 
-				bot.dialog( 'image', function( dialog ) {
-					dialog.setValueOf( 'info', 'src', '_assets/foo.png' );
-					dialog.getButton( 'ok' ).click();
-
-					assert.sameData(
-						'cke_image_resizer',
-						editor.editable().getElementsByTag( 'span' ).$[ 2 ].outerHTML.match( /(cke_image_resizer)/g )[ 0 ]
-					);
+				editor.setData( '<img src="_assets/foo.png" alt="" />', function() {
+					resume( function() {
+						editor.setReadOnly( true );
```

Comment:
This test despite refactoring should still follow the same logic, which means `editor.setReadOnly( true )` is redundant here / makes the test identical like `test readOnly set to true after initialization`.